src/codewhisperer/service/recommendationHandler.ts

```diff
@@ -192,55 +184,64 @@ export class RecommendationHandler {
-                TelemetryHelper.instance.setFirstResponseRequestId(requestId)
-                if (page === 0) {
-                    TelemetryHelper.instance.setTimeToFirstRecommendation(performance.now())
-                }
-                if (nextToken === '') {
-                    TelemetryHelper.instance.setLastRequestId(requestId)
-                    TelemetryHelper.instance.setAllPaginationEndTime()
-                }
-            } else {
-                getLogger().info('Invalid Request : ', JSON.stringify(request, undefined, EditorContext.getTabSize()))
+            if (!EditorContext.validateRequest(request)) {
                 getLogger().verbose(
-                    `Invalid Request : ${JSON.stringify(request, undefined, EditorContext.getTabSize())}`
+                    'Invalid Request : ',
+                    JSON.stringify(request, undefined, EditorContext.getTabSize())
                 )
-                errorCode = `Invalid Request`
-                if (!runtimeLanguageContext.isLanguageSupported(request.fileContext.programmingLanguage.languageName)) {
-                    this.errorMessagePrompt = `${request.fileContext.programmingLanguage.languageName} is currently not supported by CodeWhisperer`
+                const languageName = request.fileContext.programmingLanguage.languageName
+                if (!runtimeLanguageContext.isLanguageSupported(languageName)) {
+                    errorMessage = `${languageName} is currently not supported by CodeWhisperer`
                 }
+                return Promise.resolve<GetRecommendationsResponse>({
+                    result: invocationResult,
+                    errorMessage: errorMessage,
+                })
+            }
+        }
+
+        try {
+            startTime = performance.now()
+            this.lastInvocationTime = startTime
+            const mappedReq = runtimeLanguageContext.mapToRuntimeLanguage(request)
+            const codewhispererPromise = pagination
+                ? client.listRecommendations(mappedReq)
+                : client.generateRecommendations(mappedReq)
+            const resp = await this.getServerResponse(
+                triggerType,
+                config.isManualTriggerEnabled,
+                page === 0,
+                codewhispererPromise
+            )
+            TelemetryHelper.instance.setSdkApiCallEndTime()
+            latency = startTime !== 0 ? performance.now() - startTime : 0
+            if ('recommendations' in resp) {
+                recommendations = (resp && resp.recommendations) || []
+            } else {
+                recommendations = (resp && resp.completions) || []
+            }
+            invocationResult = 'Succeeded'
+            TelemetryHelper.instance.triggerType = triggerType
+            TelemetryHelper.instance.CodeWhispererAutomatedtriggerType =
+                autoTriggerType === undefined ? 'KeyStrokeCount' : autoTriggerType
```

Comment:
nit: we should probably update the default here since there is no more 'KeyStrokeCount'

Reply:
ah yes, turns out this is another globale that we don't need, will remove.